### PR TITLE
Dxt non-multiple-of-four textures

### PR DESCRIPTION
**Pull request by jamesford42**
Being more explicit about Mo4 requirements for dxt textures.

### CODE REVIEW

File: MonoGame.Framework/Graphics/Texture2D.DirectX.cs [hunk 1]
```diff
@@ -403,7 +403,20 @@ internal override SharpDX.Direct3D11.Resource CreateTexture()
...
+                _format == SurfaceFormat.Dxt5)
+            {
+                if (width % 4 != 0 || height % 4 != 0)
+                {
+                    // Note that the sharpdx call immediately below this will in fact fail
+                    // in this case. However it fails with an extremely unhelpful message. 
+                    // The message we write here matches the one given by XNA.
+                    throw new System.ArgumentException("DXT compressed texture sizes must be multiples of four.");
+                }
+            }
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Texture2D.DirectX.cs]: @tgjones @KonajuGames 

Maybe this should be higher up in the common code?  I would think even if using DXT on OpenGL or Android we would still need the same validation.  Correct?

**Comment by KonajuGames** [comment]
I've got some changes to the [content pipeline for all supported texture compression types](https://github.com/KonajuGames/MonoGame/tree/etc1) that already contains these tests, but doesn't have the test on loading the texture.  There's also the Srgb DXT formats as well that should be handled.

File: MonoGame.Framework/Graphics/Texture2D.DirectX.cs (hunk 1 above)
Comment (KonajuGames) [inline MonoGame.Framework/Graphics/Texture2D.DirectX.cs]: Yes.  It is not just a SharpDX issue.

**Comment by KonajuGames** [comment]
Closing.  Should be all handled by the ETC1 branch that was merged a while back.

**Pull request closed without merging**